src/internal/connector/onedrive/collection.go

```diff
@@ -258,9 +267,38 @@ func (oc *Collection) populateItems(ctx context.Context) {
 				metaSuffix = DirMetaFileSuffix
 			}
 
-			// Fetch data for files
+			// Fetch data for the file
+			for i := 1; i <= maxRetries; i++ {
+				itemInfo, itemData, err = oc.itemReader(ctx, item)
+
+				// retry on Timeout type errors, break otherwise.
+				if err == nil || graph.IsErrTimeout(err) == nil {
+					break
+				}
+
+				if i < maxRetries {
+					time.Sleep(1 * time.Second)
+				}
+			}
+
+			if err != nil {
+				errUpdater(*item.GetId(), err)
+				return
+			}
+
+			// Fetch metadata for the file
+			// TODO(meain): refactor these multiple retry logic
 			for i := 1; i <= maxRetries; i++ {
-				itemInfo, itemData, itemMeta, err = oc.itemReader(ctx, oc.service, oc.driveID, item)
+				if !oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
+					itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
+				} else {
+					// We are still writing the metadata file but with
+					// empty permissions as we are not sure how the
+					// restore will be called.
+					itemMeta = io.NopCloser(strings.NewReader("{}"))
```

Comment:
given that we're deciding to store "empty" permissions and we now have flags for restoring and backing up permissions it feels like we should be printing a warning or some other message if a user tries to do `restore permissions(backup no permissions)`

This should be done in a separate PR if we decide to add it, but wanted to surface it here for discussion

Reply:
I guess that would be useful. I had a similar question in https://github.com/alcionai/corso/issues/2230#issuecomment-1402331725 . The problem with warning when we backup with `--disable-permissions-backup` is that we are not adding any data that as of now mentions that we avoided backing up permissions of it just didn't have any permissions.

Reply:
made #2357 to track this